### PR TITLE
Write project

### PR DESCRIPTION
**Pull request by kcleland0818**
I wrote this module project so that it would be introduced after the OOP Pillars section; that way they could at least understand the overview of the ticket feature they are to implement.

I broke this project out into 5 parts so students can incrementally work on it:

1. Create the `Ticket` abstract class. (This is just the creation of an abstract class and none of its subclasses). I figured this part should be easy enough and can be completed right after the OOP Pillars section, as they just learned about abstract classes.
2. Evolve the array of `Concert` objects into a `List` of `Concert` objects. This task can be completed after the "Collections and Generics" section. There is a project work time lesson for this task.
3. Create the 3 `Ticket` subclasses. (This is just creating the subclasses and not actually integrating them or implementing them in with the rest of the application yet). This task can be completed after the "More on Inheritance" section. There is a project work time lesson for this task.
4. Just adding the a `LocalDate` to the `Concert` class. (Still not integrating the ticket feature in with the concert application yet, but getting closer). This task can be completed after learning dates. There is a project work time lesson for this task.
5. Determine which ticket the user will buy based off the `isStudent` and `purchaseDate` parameters that will be implemented within this task. This will also have some fun date calculations. This task _could_ be completed along with task 4 or any time after, although I recommend maybe doing this after finishing the rest of the module so it is less overwhelming. I expect this task to be the "hardest" task. There is no project work time lesson for this task - just expected to be done around the end of the module.

Let me know if you think I hold the students' hands too much in these instructions. Also, given these instructions, let me know if you think we can/should include stretch goals for students. Linda suggested maybe they could add a concert with the same performer as long as the `concertDate` differs. We also threw around the idea of implementing a concert venue interface. But that might seem too much at this point. I also don't know how many students actually would implement a stretch goal anyway.

### CODE REVIEW

**Review by linda-seiter — COMMENTED** [review]
Project looks great!

Maybe add a comment that purchaseTicketSoldOutTest is for a walkup ticket?  Or add a separate test named purchaseWalkupTicket that purchases a single WalkupTicket?  It is a bit hard given all the tests to figure out which one is for the WalkupTicket unless you look at the price in the output.

File: README.md [hunk 1]
```diff
@@ -2,31 +2,1241 @@
...
+        }
+
+        // confirm the size did not increase
+        assertEquals(5, repository.getAllConcerts().size());
+    }
+
+    @Test
+    void getConcertState() {
+        // add 3 concerts
+        assertTrue(repository.add(new Concert("The Weekend", 1000)));
```
Comment (developerjaym) [inline README.md]: Really Very Nit: 'The Weeknd' (without the final e)

**Comment by developerjaym** [comment]
I think it looks good. I wish we had a Java student who could try this out and let us know if anything's confusing.

File: .gitignore
Comment (developerjaym) [inline .gitignore:1]: Is the student going to be coding in this repository or just in their Module 2 project's repository?
If here, consider adding more to the gitignore.

**Review by developerjaym — APPROVED** [review]

**Review by linda-seiter — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (kcleland0818) [reply README.md]: This looks fixed on my end. Is it fixed on yours, Jay?

File: .gitignore
Comment (kcleland0818) [reply .gitignore:1]: Nope. I think the student should either extend their Module 2 Project in that repository or fork their module 2 project repository into a new, separate repo for them to make their Mod 3 changes (preferable). I think I included that in the introduction of this README?

**Review by linda-seiter — APPROVED** [review]
assuming weekend

**Review by linda-seiter — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (developerjaym) [reply README.md]: Yep, all good now

**Review by developerjaym — APPROVED** [review]

**Review by linda-seiter — APPROVED** [review]

**Pull request merged**